controllers/articles.js

```diff
@@ -273,4 +287,121 @@ export default class ArticleController {
       });
     }
   }
+
+  /**
+ * @description - Get all articles
+ * @static
+ * @param {Object} req - the request object
+ * @param {Object} res - the response object
+ * @memberof ArticleController
+ * @returns {Object} class instance
+ */
+  static async getAllArticles(req, res) {
+    try {
+      let {
+        query: {
+          offset, limit
+        }
+      } = req;
+      offset = Number(offset) || 0;
+      limit = Number(limit) || 10;
+      const results = await Article.findAndCountAll({
+        where: {},
+        offset,
+        limit,
+        include: [
+          {
+            model: User,
+            as: 'author',
+            attributes: ['username', 'bio', 'name']
+          },
+          {
+            model: Category,
+            as: 'category',
+            attributes: ['categoryName']
+          },
+          {
+            model: Ratings,
+          }
+        ]
+      });
+      const { count } = results;
+      const meta = Paginate({ count, limit, offset });
+      return res.status(200).json({
+        success: true,
+        results,
+        ...meta
+      });
+    } catch (error) {
+      return res.status(500).json({
+        succes: false,
+        errors: ['Oops, something wrong occured.']
+      });
+    }
+  }
+
+  /**
+  * @description - Generate query for request to get articles with order
+  * @static
+  * @param {Object} type - type of order
+  * @param {Object} amount - amount of articles to get
+  * @memberof ArticleController
+  * @returns {Object} order for findAll
+  */
+  static getQuery(type, amount) {
+    const orders = {
+      ratings:
+      `
+        SELECT "Articles".*, ROUND(AVG("Ratings".rating), 1) AS avg_rating
+        FROM "Ratings"
+        JOIN "Articles" ON "Ratings"."articleId" = "Articles".id
+        GROUP BY "Articles".id, "Articles".slug
+        ORDER BY avg_rating DESC
+        LIMIT ${Number(amount) || 5}
+      `,
+      latest:
+      `
+        SELECT *
+        FROM "Articles"
+        ORDER BY "Articles"."createdAt" DESC
+        LIMIT ${Number(amount) || 5}
+      `,
+      comments:
+      `
+        SELECT "Articles".*, COUNT("Comments".comment) AS avg_count
```

Comment:
@leksyib this should be `comment_count`. We're not running an average on the count.

Reply:
Thanks I'll change that @9jaswag

Reply:
@leksyib this hasn't been changed yet

Reply:
@9jaswag  changed now